### PR TITLE
[OTA] Allow explicit values to be set for RequestorCanConsent

### PR DESCRIPTION
**Pull request by carol-apple**
#### Problem
The `RequestorCanConsent` field in the QueryImage command is based on whether local config is enabled. This is currently incorrectly checking if local config is disabled.

#### Change overview
Check whether local config is enabled instead

#### Testing
- Verified on Darwin (where local config is disabled), the `RequestorCanConsent` field is 0
- Verified that when OTA Requestor is launched with `--requestorCanConsent`, the `RequestorCanConsent` field is 1

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/app/clusters/ota-requestor/ExtendedOTARequestorDriver.cpp [hunk 1]
```diff
@@ -35,7 +35,7 @@ bool ExtendedOTARequestorDriver::CanConsent()
                ChipLogProgress(SoftwareUpdate, "Failed to get local config disabled, using as false"));
 
     // If local config is disabled, we can't consent.
-    return localConfigDisabled == false;
+    return localConfigDisabled == true;
```
Comment (Damian-Nordic) [inline src/app/clusters/ota-requestor/ExtendedOTARequestorDriver.cpp:38]: I think the previous code was OK. Only if `localConfigDisabled == false`, the requestor can consent as described in the "User consent delegation to Nodes" section of the spec. Am I missing something?

**Review by Damian-Nordic — CHANGES_REQUESTED** [review]

File: src/app/clusters/ota-requestor/ExtendedOTARequestorDriver.cpp (hunk 1 above)
Comment (carol-apple) [reply src/app/clusters/ota-requestor/ExtendedOTARequestorDriver.cpp:38]: I missed the part talking about this in the spec and reading the comment, I thought it had implemented the opposite logic. I think I'll just update the comment to better reflect the spec.

Also, it seems given certain platforms like Darwin will always have `LocalConfigDisabled = false`, that means the `RequestorCanConset` field will always be 1, whether `--requestorCanConsent` field is supplied or not. I'll update that command line option so the field can be specified explicitly when supplied.

[the remark this answers is not in the document]
Comment (isiu-apple) [reply src/app/clusters/ota-requestor/ExtendedOTARequestorDriver.cpp:38]: (This comment was in pending state for a while and I just got it out of pending state.  I see the discussion is on-going below so just ignore this one.)

**Review by shubhamdp — APPROVED** [review]

File: examples/ota-requestor-app/linux/main.cpp [hunk 2]
```diff
@@ -173,7 +174,14 @@ bool HandleOptions(const char * aProgram, OptionSet * aOptions, int aIdentifier,
         gPeriodicQueryTimeoutSec = static_cast<uint32_t>(strtoul(aValue, NULL, 0));
         break;
     case kOptionRequestorCanConsent:
-        gRequestorCanConsent.SetValue(true);
+        if (strcmp(aValue, "true") == 0)
```
Comment (isiu-apple) [inline examples/ota-requestor-app/linux/main.cpp:177]: What if user had a typo and aValue here was neither `true` nor `false`.  Shall we give feedback that it was an invalid argument?

Comment (carol-apple) [reply examples/ota-requestor-app/linux/main.cpp:177]: Added an error

**Review by Damian-Nordic — APPROVED** [review]

**Pull request merged**